examples/custom_shader/custom_pass.rs

```diff
@@ -0,0 +1,348 @@
+        encoder.bind_graphics_pipeline(&self.pipeline);
+
+        // Bind the Dynamic buffer with the scale to the encoder
+        self.env.bind(index, &self.pipeline_layout, 0, &mut encoder);
+
+        // Bind the vertex buffer to the encoder
+        self.vertex.bind(index, 0, 0, &mut encoder);
+
+        // Draw the vertices
+        unsafe {
+            encoder.draw(0..self.vertex_count as u32, 0..1);
+        }
+    }
+
+    fn dispose(self: Box<Self>, factory: &mut Factory<B>, _world: &World) {
+        unsafe {
+            factory.device().destroy_graphics_pipeline(self.pipeline);
+            factory
+                .device()
+                .destroy_pipeline_layout(self.pipeline_layout);
+        }
+    }
+}
+
+fn build_custom_pipeline<B: Backend>(
+    factory: &Factory<B>,
+    subpass: hal::pass::Subpass<'_, B>,
+    framebuffer_width: u32,
+    framebuffer_height: u32,
+    layouts: Vec<&B::DescriptorSetLayout>,
+) -> Result<(B::GraphicsPipeline, B::PipelineLayout), failure::Error> {
+    let pipeline_layout = unsafe {
+        factory
+            .device()
+            .create_pipeline_layout(layouts, None as Option<(_, _)>)
+    }?;
+
+    // Load the shaders
+    let shader_vertex = unsafe { VERTEX.module(factory).unwrap() };
+    let shader_fragment = unsafe { FRAGMENT.module(factory).unwrap() };
+
+    // Build the pipeline
+    let pipes = PipelinesBuilder::new()
+        .with_pipeline(
+            PipelineDescBuilder::new()
+                // This Pipeline uses our custom vertex description and does not use instancing
+                .with_vertex_desc(&[(CustomArgs::vertex(), pso::VertexInputRate::Vertex)])
+                .with_input_assembler(pso::InputAssemblerDesc::new(hal::Primitive::TriangleList))
+                // Add the shaders
+                .with_shaders(util::simple_shader_set(
+                    &shader_vertex,
+                    Some(&shader_fragment),
+                ))
+                .with_layout(&pipeline_layout)
+                .with_subpass(subpass)
+                .with_framebuffer_size(framebuffer_width, framebuffer_height)
+                // We are using alpha blending
+                .with_blend_targets(vec![pso::ColorBlendDesc(
+                    pso::ColorMask::ALL,
+                    pso::BlendState::ALPHA,
+                )]),
+        )
+        .build(factory, None);
+
+    // Destoy the shaders once loaded
+    unsafe {
+        factory.destroy_shader_module(shader_vertex);
+        factory.destroy_shader_module(shader_fragment);
+    }
+
+    // Handle the Errors
+    match pipes {
+        Err(e) => {
+            unsafe {
+                factory.device().destroy_pipeline_layout(pipeline_layout);
+            }
+            Err(e)
+        }
+        Ok(mut pipes) => Ok((pipes.remove(0), pipeline_layout)),
+    }
+}
+
+/// A [RenderPlugin] for our custom plugin
+#[derive(Default, Debug)]
+pub struct RenderCustom {
+    target: Target,
```

Comment:
This target is never exposed or changed in any way. There usually is not much point in allowing such configuration in custom render plugins (because those are tailored to your needs anyway), so i would recommend hardcoding `Target::Main` directly to `extend_target` call and get rid of this value. It is of course not really wrong, but here it's incomplete and not really necessary.

Reply:
Removed this.